packetbeat/protos/nfs/rpc.go

```diff
@@ -187,6 +186,45 @@ func getRPCConnection(private protos.ProtocolData) *rpcConnectionData {
 }
 
 // Parse function is used to process TCP payloads.
+//
+// From RFC 5531:
+//
+// 11. Record Marking Standard
+// When RPC messages are passed on top of a byte stream transport
+// protocol (like TCP), it is necessary to delimit one message from
+// another in order to detect and possibly recover from protocol errors.
+// This is called record marking (RM).  One RPC message fits into one RM
+// record.
+//
+// A record is composed of one or more record fragments.  A record
+// fragment is a four-byte header followed by 0 to (2**31) - 1 bytes of
+// fragment data.  The bytes encode an unsigned binary number; as with
+// XDR integers, the byte order is from highest to lowest.  The number
+// encodes two values -- a boolean that indicates whether the fragment
+// is the last fragment of the record (bit value 1 implies the fragment
+// is the last fragment) and a 31-bit unsigned binary value that is the
+// length in bytes of the fragment's data.  The boolean value is the
+// highest-order bit of the header; the length is the 31 low-order bits.
+// (Note that this record specification is NOT in XDR standard form!)
+func parseFragment(rawData []byte) (partialMessage []byte, fragmentSize int, isLast bool) {
+	if len(rawData) < 4 {
+		debugf("Waiting for more data")
+		return nil, 0, false
+	}
+
+	header := binary.BigEndian.Uint32(rawData[0:4])
+	fragmentSize = 4 + int(header&rpcSizeMask)
+	isLast = (header & rpcLastFrag) != 0
+
+	if len(rawData) < fragmentSize {
+		debugf("Waiting for more data")
+		return nil, 0, false
+	}
+
+	partialMessage = rawData[4:fragmentSize]
```

Comment:
```suggestion
	partialMessage = rawData[4:fragmentSize - 4]
```
Unless I'm missing something:
fragmentSize includes the header, partialMessage skips over the header, so it should subtract it.

Reply:
I thought it was size, please disregard :)